arch/risc-v/src/mpfs/mpfs_mm_init.c

```diff
@@ -40,6 +40,23 @@
  * Pre-processor Definitions
  ****************************************************************************/
 
+/* Ensure the sections are aligned properly, requirement is 2MB due to the
+ * L3 page table size (one table maps 2MB of memory). This mapping cannot
+ * handle unaligned L3 sections.
+ */
+
+#if (KFLASH_START & RV_MMU_SECTION_ALIGN) != 0
+#  error "Kernel flash start address is not aligned to section boundary (2MB)"
+#endif
+
+#if (KSRAM_START & RV_MMU_SECTION_ALIGN) != 0
+#  error "Kernel RAM start address is not aligned to section boundary (2MB)"
+#endif
+
+#if (PGPOOL_START & RV_MMU_SECTION_ALIGN) != 0
+#  error "Page pool is not aligned to section boundary (2MB)"
+#endif
```

Comment:
```suggestion
static_assert((KFLASH_START & RV_MMU_SECTION_ALIGN) == 0,
              "Kernel flash start address is not aligned to section boundary (2MB)");

static_assert((KSRAM_START & RV_MMU_SECTION_ALIGN) == 0,
              "Kernel RAM start address is not aligned to section boundary (2MB)");

static_assert((PGPOOL_START & RV_MMU_SECTION_ALIGN) == 0,
              "Page pool is not aligned to section boundary (2MB)");
```

Reply:
Good suggestion but it will not help here, these must be evaluated at run time unfortunately, because `KFLASH_START ` etc. are resolved at link time via relocation symbols and static_assert AFAIK is a compile time test.

in boards/risc-v/mpfs/icicle/include/board_memorymap.h:
```
extern uint8_t          __kflash_start[];
#define KFLASH_START    (uintptr_t)__kflash_start
```

And so forth. This was the mistake I was talking about before, testing these must be done run-time.